include/swift/AST/ASTMangler.h

```diff
@@ -318,6 +318,9 @@ class ASTMangler : public Mangler {
                       const ValueDecl *forDecl = nullptr);
   void appendFunctionType(AnyFunctionType *fn, bool isAutoClosure = false,
                           const ValueDecl *forDecl = nullptr);
+  void appendClangType(AnyFunctionType *fn);
+  template <typename FnType>
+  void appendClangType(FnType *fn, llvm::raw_svector_ostream &os);
```

Comment:
What do you think about replacing `FnType` with `ClangTypeInfo`? This way it could potentially work for more types and the API is a bit more explicit.

Reply:
I'm passing in a function type primarily because we need access to the ASTContext in the method's implementation, and ASTMangler doesn't carry an ASTContext. Is there a way to get the ASTContext directly from the ASTMangler?

Reply:
That's fair. I think you can use `Mod->getASTContext()`, but I don't feel strongly. If you'd rather the template is OK with me.

Reply:
In practice, sometimes the `Mod` pointer is null (I haven't checked when exactly), so the tests crash if I do that.